stetho/src/main/java/com/facebook/stetho/server/LocalSocketHttpServer.java

```diff
@@ -190,32 +188,28 @@ private static String getProcessName() throws IOException {
     try {
       int n = stream.read(cmdlineBuffer);
       success = true;
-      int endIndex = tidyProcessName(cmdlineBuffer, n);
+      int endIndex = findProcessNameEndIndex(cmdlineBuffer, n);
+      if (endIndex < 0) {
+        throw new IOException("Unexpected cmdline buffer: " + new String(cmdlineBuffer, 0, n));
+      }
       return new String(cmdlineBuffer, 0, endIndex);
     } finally {
       Util.close(stream, !success);
     }
   }
 
-  private static int tidyProcessName(byte[] processName, int count) {
-    int i = 0;
-    OUTER:
-    for (; i < count; i++) {
-      switch (processName[i]) {
-        case '\\':
-        case '.':
-        case ':':
-          processName[i] = '_';
-          break;
+  private static int findProcessNameEndIndex(byte[] cmdline, int count) {
```

Comment:
That's what this diff is intending to do.  It no longer does replacements at all.  If it applies first, the one which uses regexps will be rebased and the regexps will be removed.  If this applies last, the regexp will be replaced with nothing...

I developed the diffs in such a way that they were both intentionally based off `master` so neither had to go first strictly.